src/sysc/scc/tracer.cpp

```diff
@@ -55,7 +55,7 @@ tracer::tracer(std::string const&& name, file_type tx_type, file_type sig_type,
     if(sig_type == ENABLE)
         sig_type = static_cast<file_type>(sig_trace_type.get_value());
     if(sig_type != NONE) {
-        switch(sig_type) {
+        switch(tx_type) {
```

Comment:
This logic is wrong as it effectively skips the signal trace type. This way it falls pacl to vcd if selected any of the tx trace type or it falls back to text type transaction trace if any of the signal trace types is selected.